src/Santa/MessageDispatcher.php

```diff
@@ -29,15 +29,17 @@ class MessageDispatcher
     public function dispatchRemainingMessages(SecretSanta $secretSanta, ApplicationInterface $application): void
     {
         $startTime = time();
+        $failedUsers = $secretSanta->getErrors();
 
         foreach ($secretSanta->getRemainingAssociations() as $giver => $receiver) {
-            if ((time() - $startTime) > 5) {
-                throw new MessageDispatchTimeoutException($secretSanta);
-            }
-
-            $application->sendSecretMessage($secretSanta, $giver, $receiver);
+            if (!\array_key_exists($giver, $failedUsers)) {
```

Comment:
early ~return~ continue, please :smile:

Reply:
Never sure about how to fix it, is it better now or still wrong ?

Reply:
I don't understand your change. The goal is to avoid a new level of indentation. If the current giver is failing, just `continue` to skip it